playbooks/bios-upgrade-prod-dell.yaml

```diff
@@ -0,0 +1,74 @@
+- hosts: nerc_ocp_prod_workers
+  gather_facts: false
+  vars:
+    step_firmware: 2.70.70.70
+    target_firmware: 2.83.83.83
+    target_bios: 2.13.0
+    drm_server: 10.255.0.25
+    step_firmware_dir: fc430-step
+    target_firmware_dir: fc430
+    step_firmware_catalog: Catalog.xml
+    target_firmware_catalog: fc430_1.10_Catalog.xml
+
+  tasks:
+    - name: Get BMC IP
+      set_fact:
+        bmc_ip: "{{ lookup ('dig', '{{ bmc_addr }}') }}"
+
+    - name: Get firmware versions
+      delegate_to: localhost
+      dellemc.openmanage.idrac_firmware_info:
+        idrac_ip: "{{ bmc_ip }}"
+        idrac_user: "{{ bmc_user }}"
+        idrac_password: "{{ bmc_password }}"
+        validate_certs: false
+      register: current_firmware
+
+    - name: Display current bios major firmware
+      set_fact:
+        bios_ver: "{{ item }}"
+      loop: "{{ current_firmware | json_query(firmware_query) }}"
+      vars:
+        firmware_query: "firmware_info.Firmware[?ComponentType=='BIOS'].VersionString"
+
+    - name: Display current bios major firmware
+      set_fact:
+        idrac_ver: "{{ item }}"
+      loop: "{{ current_firmware | json_query(firmware_query) }}"
+      vars:
+        firmware_query: "firmware_info.Firmware[?ElementName=='Integrated Dell Remote Access Controller'].VersionString"
+
+    - name: Update to step firmware if required
+      delegate_to: localhost
+      dellemc.openmanage.idrac_firmware:
+        idrac_ip: "{{ bmc_ip }}"
+        idrac_user: "{{ bmc_user }}"
+        idrac_password: "{{ bmc_password }}"
+        validate_certs: false
+        job_wait: true
+        share_name: "http://{{ drm_server }}/{{ step_firmware_dir }}/"
+        catalog_file_name: "{{ step_firmware_catalog }}"
+        reboot: true
+      when: idrac_ver < step_firmware
+
+    - name: Update to target firmware if required
+      delegate_to: localhost
+      dellemc.openmanage.idrac_firmware:
+        idrac_ip: "{{ bmc_ip }}"
+        idrac_user: "{{ bmc_user }}"
+        idrac_password: "{{ bmc_password }}"
+        validate_certs: false
+        job_wait: true
+        share_name: "http://{{ drm_server }}/{{ target_firmware_dir }}/"
+        catalog_file_name: "{{ target_firmware_catalog }}"
+        reboot: true
+      when: idrac_ver < target_firmware
+
+    - name: Disable host header check
+      delegate_to: localhost
+      become: true
+      raw: >-
```

Comment:
If we're targeting `localhost`, this should probably be `command` rather than `raw` (we only need `raw` when executing against a remote device that doesn't have Python).

Can you either (a) add some docs re: how to install the `racadm` command locally, or (b) just rewrite this to run `racadm` directly on the idrac? E.g.:

```
- name: Disable host header check
  delegate_to: "{{ bmc_addr }}"
  raw: >-
    racadm set idrac.webserver.hostheadercheck 0
  register: result
  failed_when: >-
    "Object value modified successfully" not in result.stdout
```

Reply:
I was having issues with running directly on the idrac which is why I moved it to local, I can tinker with it some more though.

Reply:
We can merge this and fix it in a future PR. I was mostly concerned about the dependency on a locally installed `racadm` command, and it's fine if that's something we address in documentation.